packages/flutter_tools/lib/src/asset.dart

```diff
@@ -912,8 +912,17 @@ class ManifestAssetBundle implements AssetBundle {
     Package? attributedPackage,
     List<String>? flavors,
   }) {
-    final String directoryPath = _fileSystem.path.join(
-        assetBase, assetUri.toFilePath(windows: _platform.isWindows));
+    final String directoryPath = (() {
+      try {
+        return _fileSystem.path
+          .join(assetBase, assetUri.toFilePath(windows: _platform.isWindows));
+      } on UnsupportedError catch (e) {
+        throwToolExit(
+          'Unable to search for asset files in directory path "${Uri.decodeFull(assetUri.path)}". '
```

Comment:
Could Uri.decodeFull throw?

Reply:
I can remove this, but the user will see Uri-encoded paths in the error message, which may still be sufficient

Reply:
Well, if we tried to throw a tool exit but in constructing the message threw something else, that would be a bad experience, right? or is that something we wouldn't want to tool exit for?

Reply:
Addressed in [prevent potential crash when trying to construct exit message](https://github.com/flutter/flutter/pull/141214/commits/35a77eb640e01f762b782cf8d2ed733dafa4f469)